test_util/launch_configs/onprem-from-info.yaml

```diff
@@ -0,0 +1,15 @@
+---
+# no deployment_name provided because it was set in the cloud_cluster
+deployment_type: onprem
+installer_url: https://downloads.dcos.io/dcos/testing/master/dcos_generate_config.sh
+provider: cloud_cluster
+provider_info_json_path: /foo/bar/important/path
+dcos_config:
+    num_private_agents: 2
+    num_public_agents: 1
+    num_masters: 3
+    cluster_name: My Awesome DC/OS
+    resolvers:
+        - 8.8.4.4
+        - 8.8.8.8
+    security: strict
```

Comment:
Instead of adding new options like `num_masters` mixed in with standard config.yaml, is there any way we could have this section map 1 to 1 with our existing installer config format and just make some incompatible options disallowed and infer the rest from the provider info or via configuration at an earlier provision stage? I'm concerned that this is going to get really confusing, since the existing config.yaml format is already very complicated.